File: sdk/py/tutorials/ml/nlp/tu_nlp_byte_mamba3.py
#!/usr/bin/env python3
"""OA Tutorial — Byte Mamba-3 · all-position LM (autograd).

Mirrors sdk/rs/tutorials/ml/nlp/tu_nlp_byte_mamba3.rs.
"""

import sys
sys.path.insert(0, str(__import__("pathlib").Path(__file__).parent))
import _nlp_common as nlp
import oa


def main() -> None:
	engine = oa.Engine()
	model = oa.ml.ByteMamba3(engine)
	optimizer = oa.ml.AdamW(model.all_parameters(), learning_rate=0.003)
	nlp.run_byte("Byte Mamba-3 · all-position LM", model, optimizer, engine)


if __name__ == "__main__":
	main()

File: sdk/rs/tutorials/ml/nlp/tu_nlp_byte_mamba3.rs
mod _byte_common;

fn main() -> oa::Result<()> {
	_byte_common::run(
		_byte_common::Tutorial {
			title: "OA Tutorial — Byte Mamba-3 · all-position LM (autograd)",
			description: "Byte Embedding → Mamba-3(32,state=32,expand=2) + residual → Linear(32→256)",
			timer_name: "byte_mamba3_training_step",
			checkpoint_stem: "oars_byte_mamba3",
			learning_rate: 0.003,
			expected_loss: oa::sdk::ml::nlp::BYTE_MAMBA3_FINAL_LOSS,
			expected_accuracy: oa::sdk::ml::nlp::BYTE_MAMBA3_ACCURACY,
			expected_generation: oa::sdk::ml::nlp::BYTE_MAMBA3_REFERENCE_GENERATION,
		},
		oa::sdk::ml::nlp::ByteMamba3::new,
	)
}
